src/app/component/projects/pages/assign/assign.component.html

```diff
@@ -0,0 +1,49 @@
+<app-wip>
+  <ng-container *ngIf="ready">
+    <h1>{{ project.name }}</h1>
+
+    <h2>Assign Sites</h2>
+
+    <p>
+      Select the sites to be part of this project, and click 'Update' at the
+      bottom of this page.
+    </p>
+    <p>Only the selected sites will be part of this project.</p>
+
+    <ngx-datatable
+      #table
+      class="bootstrap"
+      [columnMode]="ColumnMode.force"
+      [columns]="columns"
+      [draggable]="false"
+      [footerHeight]="50"
+      [headerHeight]="50"
+      [limit]="25"
+      [reorderable]="false"
+      [rowHeight]="'auto'"
+      [rows]="rows"
+      [scrollbarH]="true"
+      [selectAllRowsOnPage]="false"
+      [selected]="selected"
+      [selectionType]="SelectionType.checkbox"
+      (select)="onSelect($event)"
```

Comment:
Is it possible to move these attributes to the controller code? For so many options it would seem neater to just `...options` rather than do this...

Reply:
The most angular method for this would be to create a wrapper component which replicates the inputs from `ngx-datatable` and sets defaults for the inputs not provided. I intend to do this once I've built enough tables to know a) what the defaults are, and b) link up the tables to changing data so I can test updating the tables. For now I've created an abstract class which defines a list of defaults to reduce the amount of boilerplate and magic numbers.

Reply:
This is basically what I'm referring too: https://github.com/angular/angular/issues/14545
Looks like it is not a thing 🤷‍♂️

Don't over nest components - it just creates more work for browser and angular.

Maybe a custom directive would be better, like `ngx-standard-options` that does the binding that way.